lib/beaker/host/unix/exec.rb

```diff
@@ -187,17 +187,22 @@ def ssh_service_restart
   def ssh_permit_user_environment
     case self['platform']
     when /debian|ubuntu|cumulus/
-      exec(Beaker::Command.new("echo '\nPermitUserEnvironment yes' >> /etc/ssh/sshd_config"))
+      exec(Beaker::Command.new("echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/sshd_config.new"))
+      exec(Beaker::Command.new("mv /tmp/sshd_config.new /etc/ssh/sshd_config"))
     when /el-7|centos-7|redhat-7|oracle-7|scientific-7|eos-7/
-      exec(Beaker::Command.new("echo '\nPermitUserEnvironment yes' >> /etc/ssh/sshd_config"))
+      exec(Beaker::Command.new("echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/sshd_config.new"))
+      exec(Beaker::Command.new("mv /tmp/sshd_config.new /etc/ssh/sshd_config"))
     when /el-|centos|fedora|redhat|oracle|scientific|eos/
-      exec(Beaker::Command.new("echo '\nPermitUserEnvironment yes' >> /etc/ssh/sshd_config"))
+      exec(Beaker::Command.new("echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/sshd_config.new"))
+      exec(Beaker::Command.new("mv /tmp/sshd_config.new /etc/ssh/sshd_config"))
     when /sles/
-      exec(Beaker::Command.new("echo '\nPermitUserEnvironment yes' >> /etc/ssh/sshd_config"))
+      exec(Beaker::Command.new("echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/sshd_config.new"))
+      exec(Beaker::Command.new("mv /tmp/sshd_config.new /etc/ssh/sshd_config"))
     when /solaris/
       # kept solaris here because refactoring it into its own Host module
       # conflicts with the solaris hypervisor that already exists
-      exec(Beaker::Command.new("echo '\nPermitUserEnvironment yes' >> /etc/ssh/sshd_config"))
+      exec(Beaker::Command.new("echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/sshd_config.new"))
+      exec(Beaker::Command.new("mv /tmp/sshd_config.new /etc/ssh/sshd_config"))
     when /(free|open)bsd/
       exec(Beaker::Command.new("sudo perl -pi -e 's/^#?PermitUserEnvironment no/PermitUserEnvironment yes/' /etc/ssh/sshd_config"), {:pty => true} )
```

Comment:
I was also a little put off by the fact every command for each case statement except for *bsd is the same...